lib/mongo/operation/shared/response_handling.rb

```diff
@@ -132,7 +132,10 @@ def maybe_add_retryable_write_error_label!(error, client, session)
         in_transaction = session && session.in_transaction?
         committing_transaction = in_transaction && session.committing_transaction?
         aborting_transaction = in_transaction && session.aborting_transaction?
-        retry_writes = client && client.options[:retry_writes]
+        modern_retry_writes = client && client.options[:retry_writes]
+        legacy_retry_writes = client && !client.options[:retry_writes] &&
+          client.options[:max_write_retries] != 0
```

Comment:
Can this be `> 0` please?

Reply:
I think it should actually be `client.max_write_retries > 0` because the method assigns a default. Good call.